sycl/include/sycl/ext/oneapi/sub_group_mask.hpp

```diff
@@ -283,6 +313,14 @@ struct sub_group_mask {
   }
 
 private:
+  static size_t GetMaxLocalRangeSize() {
+#ifdef __SYCL_DEVICE_ONLY__
+    return __spirv_SubgroupMaxSize();
+#else
+    return max_bits;
```

Comment:
@Pennycook - What would the right behavior for the "max local range" be on host?

Reply:
Good question.  I'm not sure -- we deliberately said the number of bits was implementation-defined, and it's unclear to me what mask size(s) we would want to support on a host device.

SIMD64 is _probably_ a good bet for the widest we'd want to go: 512-bit SIMD / 8-bit data types = 64, and 64 is the biggest size we can support without moving to an unusual mask representation (like `vec<int,4>`).  So, maybe 64?

Reply:
@Pennycook Increased the size to 64 on host. Should it be described in the spec? bc6ca62

Reply:
Thanks for updating this.  I don't think we need to change the spec, because it says:

> The group mask type is an opaque type, permitting implementations to use any mask representation

...so the fact we use 64 bits is an implementation detail.

But I do think we need to change this part of the implementation:
https://github.com/intel/llvm/blob/c3af751d5430f8a2288c4f9ecb46087e0c13723b/sycl/include/sycl/ext/oneapi/sub_group_mask.hpp#L39-L44

...because the full quote from the specification says (emphasis mine):
> The group mask type is an opaque type, permitting implementations to use any mask representation **that has the same size and alignment across host and device**.

If we're using 64 bits for AMD devices and the host, we also need to use 64 bits for our other devices (even if that means we don't use the upper 32 bits some of the time).

Reply:
So, we should use 64 bits for all cases then, right?

Reply:
Yes, I think so.